python/ray/tests/conftest.py

```diff
@@ -855,7 +870,20 @@ def start_cluster(ray_start_cluster_enabled, request):
     else:
         address = cluster.address
 
-    yield cluster, address
+    return cluster, address
+
+
+# Used to test both Ray Client and non-Ray Client codepaths.
+# Usage: In your test, call `ray.init(address)`.
+@pytest.fixture(scope="function", params=["ray_client", "no_ray_client"])
+def start_cluster(ray_start_cluster_enabled, request):
+    yield _start_cluster(ray_start_cluster_enabled, request)
+
+
+# Same as `start_cluster` but module-scoped.
+@pytest.fixture(scope="module", params=["ray_client", "no_ray_client"])
+def start_cluster_shared(ray_start_cluster_enabled_shared, request):
```

Comment:
Can we use both `start_cluster` and `start_cluster_shared` in the same test file?

Reply:
They shouldn't be; let me add enforcement for this.